src/main/java/uk/gov/hmcts/ccd/domain/service/accessprofile/filter/FilterRoleAssignmentsImpl.java

```diff
@@ -0,0 +1,139 @@
+            roleAssignment.getId());
+        Optional<SecurityClassification> securityClassification = SecurityClassificationUtils
+            .getSecurityClassification(roleAssignment
+                .getClassification());
+        if (securityClassification.isPresent()) {
+            boolean value = caseHasClassificationEqualOrLowerThan(SecurityClassificationUtils
+                .getSecurityClassification(roleAssignment
+                    .getClassification())
+                .get())
+                .test(caseDetails);
+            roleAssignment.getMatchingResults().setValidClassification(value);
+        }
+        return roleAssignment.getMatchingResults().isValidClassification();
+    }
+
+    private boolean validateCaseId(CaseDetails caseDetails, RoleAssignment roleAssignment) {
+        log.debug("Apply filter on case id {} for role assignment {}",
+            roleAssignment.getAttributes().getCaseId(),
+            roleAssignment.getId());
+        roleAssignment.getMatchingResults().setValidCaseId(
+            roleAssignment.getAttributes().getCaseId().equals(caseDetails.getReferenceAsString()));
+        return roleAssignment.getMatchingResults().isValidCaseId();
+    }
+
+    private boolean validateLocation(RoleAssignment roleAssignment) {
+        log.debug("Apply filter on location {} for role assignment {}",
+            roleAssignment.getAttributes().getLocation(),
+            roleAssignment.getId());
+        boolean isEmpty = StringUtils.isEmpty(roleAssignment.getAttributes().getLocation());
+        if (isEmpty) {
+            roleAssignment.getMatchingResults().setValidLocation(true);
+        } else {
+            roleAssignment.getMatchingResults().setValidLocation(
+                roleAssignment.getAttributes().getLocation()
+                    .equals("" /* location */));
+        }
+        return roleAssignment.getMatchingResults().isValidLocation();
+    }
+
+    private boolean validateRegion(RoleAssignment roleAssignment) {
+        log.debug("Apply filter on region {} for role assignment {}",
+            roleAssignment.getAttributes().getRegion(),
+            roleAssignment.getId());
+        boolean isEmpty = StringUtils.isEmpty(roleAssignment.getAttributes().getRegion());
+        if (isEmpty) {
+            roleAssignment.getMatchingResults().setValidRegion(true);
+        } else {
+            roleAssignment.getMatchingResults().setValidRegion(
+                roleAssignment.getAttributes().getRegion()
+                    .equals("" /* region */));
+        }
+        return roleAssignment.getMatchingResults().isValidRegion();
+    }
+
+    private boolean validateStartAndEndDate(RoleAssignment roleAssignment) {
+        log.debug("Apply filter on start {} and end time {} for role assignment {}",
+            roleAssignment.getBeginTime(),
+            roleAssignment.getEndTime(),
+            roleAssignment.getId());
+        LocalDateTime now = LocalDateTime.now();
```

Comment:
can we inject a Clock as done here please: https://github.com/hmcts/ccd-data-store-api/blob/7463f215c020c2c8d9c607c8b856b9de5a55a362/src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/CreateCaseEventService.java#L100
it will simplify testing

Reply:
will discuss this with Rafal

Reply:
https://docs.oracle.com/javase/8/docs/api/java/time/package-summary.html
Looks like there is nothing about the format of the Clock when comes to serialization. We need ISO-8601. Digging further looks like Jackson jsr-310 extension does not support it:
https://fasterxml.github.io/jackson-modules-java8/javadoc/datetime/2.12/com/fasterxml/jackson/datatype/jsr310/JavaTimeModule.html
https://stackoverflow.com/questions/27952472/serialize-deserialize-java-8-java-time-with-jackson-json-mapper

We could provide a @Qualifier("utcLocalDateTime") the same way you are suggesting a Clock. Or we can use the Clock and pass it to the LocalDateTime, because all now() overrides use a Clock at the end.

// LocalDateTime.java
    public static LocalDateTime now(Clock clock) {
        Objects.requireNonNull(clock, "clock");
        final Instant now = clock.instant();  // called once
        ZoneOffset offset = clock.getZone().getRules().getOffset(now);
        return ofEpochSecond(now.getEpochSecond(), now.getNano(), offset);
    }

I am afraid LocalDateTime.now(); will get the default zone in the LocalDateTime.java
    public static LocalDateTime now() {
        return now(Clock.systemDefaultZone());
    }
    but we could use
    public static LocalDateTime now(ZoneId zone) {
        return now(Clock.system(zone));
    }
    
    Because we know the roleAssignments are in UTC I would use:
    LocalDateTime now = LocalDateTime.now(Clock.systemUTC());
    
    https://docs.oracle.com/javase/8/docs/api/java/time/Clock.html#systemUTC--